FlowCrypt/Controllers/SignIn/SignInViewController.swift

```diff
@@ -19,19 +21,22 @@ final class SignInViewController: BaseViewController {
 
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
-        navigationController?.isNavigationBarHidden = true
+        navigationController?.setNavigationBarHidden(true, animated: false)
     }
     
     override func viewWillDisappear(_ animated: Bool) {
         super.viewWillDisappear(animated)
-        navigationController?.isNavigationBarHidden = false
+        navigationController?.setNavigationBarHidden(false, animated: false)
     }
 
     // MARK: - Events
     @IBAction func signInWithGmailButtonPressed(_ sender: Any) {
-        async({ try await(GoogleApi.instance.signIn(viewController: self)) }, then: { [weak self] user in
-            self?.performSegue(withIdentifier: "RecoverSegue", sender: nil)
-        })
+        showSpinner()
+        googleAPI.signIn(viewController: self)
+            .then(on: .main) { [weak self] _ in
```

Comment:
when `googleAPI.signin()` returns a failed promise, how would that be handled?

Reply:
From the code, it seems it would get stuck with a spinner running forever.

Reply:
It should be somehow handled. But it wasn't handled before, so it's up to you. 
I will just remove spinner for now(see next commit, it still `wip`)

Reply:
It was handled before, every time any promise was used.

```Swift
        async({ try await(GoogleApi.instance.signIn(viewController: self)) }, then: { [weak self] user in
            self?.performSegue(withIdentifier: "RecoverSegue", sender: nil)
        })
```
This will call a method that has a default `fail` to hide spinner and show error alert, I think.

Reply:
If you don't like the `async` convenience thing for any reason, you'll have to make sure to manually decide what to do every time there's an error. Most of the time, it's appropriate to tell the user - that's why I had a method for it that does that.

Reply:
Yes but in this code you have a lot of memory leaks. 
self was captured in at least one time. 
Also in case of self is already nil async operation will cause app to crash

Reply:
Sure, I'm happy for all of that to get cleaned up.

In the previous code, all promise errors were handled, and this is something to be careful to preserve after the refactor, too.

Here's the async method it was calling: 

```Swift
    func async<T>(_ work: @escaping () throws -> T, then: @escaping (T) throws -> Void, fail alertMsg: String = "Action failed") {
        self.async(work, then: then, fail: { error in self.showErrAlert("\(alertMsg)\n\n \(error)") })
    }
```
Notice the default error handler. `showErrAlert` will also hide spinner, so this will be very convenient in 99% of error situations.

Everywhere you do a `.then`, you _have_  to do `.catch`, else you're silently dropping errors. 

So it would look something like:
```Swift
        googleAPI.signIn(viewController: self)
            .then(on: .main) { [weak self] result in ... }
            .catch(on: .main) { [weak self] error in self.showErrAlert("\(alertMsg)\n\n \(error)") }
```
as an example.

Reply:
This `async` func has side effects which better to avoid. 
Also better to use composition over inheritance, so try not to use BaseViewControllers and so on
Same with retain cycles/memory leaks. It will cause your app to crashes

Reply:
I agree with all of that. Please do it in a way that no promise rejections are silently dropped :+1: